helios-services/src/main/java/com/spotify/helios/servicescommon/ZooKeeperRegistrarService.java

```diff
@@ -51,7 +52,7 @@
 
   private final ZooKeeperClient client;
   private final ZooKeeperRegistrar zooKeeperRegistrar;
-  private final CountDownLatch zkRegistrationSignal;
+  private final Optional<CountDownLatch> zkRegistrationSignal;
```

Comment:
Would we ever allow this to be absent? If not, perhaps just `checkNotNull()` in the constructor and remove the Optional?